### PR TITLE
WIP: DragDropDevice: Ignore raw events if window is disabled on platform

### LINKED ISSUE #5940: Drag & drop should be disabled if a modal window open
**Describe the bug**
When a dialog is open other windows should not handle for drag and drop events.

It causes problems when you already drop something and opened a dialog, but then the user drops something into a non dialog window again. Another instance instance of the same dialog pop ups.

**To Reproduce**

```csharp
public class App : Application
{
    [STAThread]
    public static void Main(string[] args) => AppBuilder
        .Configure<App>()
        .UsePlatformDetect()
        .StartWithClassicDesktopLifetime(args);

    public override void Initialize()
    {
        AvaloniaXamlLoader.Load(this);
    }

    public override void OnFrameworkInitializationCompleted()
    {
        if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
        {
            desktop.MainWindow = new ThrowOnDrop();
        }

        base.OnFrameworkInitializationCompleted();
    }

    class ThrowOnDrop : Window
    {
        public ThrowOnDrop()
        {
            SetValue(DragDrop.AllowDropProperty, true);
            AddHandler(DragDrop.DragEnterEvent, (s, e) => throw new Exception("No!"));
        }

        protected override void OnOpened(EventArgs e)
        {
            new Window()
                .ShowDialog(this);
            base.OnOpened(e);
        }
    }
}
```

**Expected behavior**
Only opened dialog should handle drag and drop, but other windows should not do that.

**Desktop (please complete the following information):**

- OS: Windows

### PR DESCRIPTION
**Pull request by mat1jaczyyy**
## What does the pull request do?

This PR ignores dragdrop events on disabled windows.

## What is the current behavior?

As described in #5940, drag and drop doesn't get disabled for parents of dialogs. 

## What is the updated/expected behavior with this PR?

Generally, any window that is disabled on platform should have drag drop events ignored.

## How was the solution implemented (if it's not obvious)?

Added a property to IWindowImpl that checks platform enabled status, accessible via IInputRoot. DragDropDevice checks for this, and ignores events if the root is disabled.

I'm unsure about how to write tests for this. X11/Linux implementation needs checking, and someone needs to do macOS implementation (so this isn't mergeable yet)

## Checklist

- [ ] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [x] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues
Fixes #5940

### CODE REVIEW

**Comment by Gillibald** [comment]
Is it possible to disable a `TopLevel` control? Is D&D only available for windows? Should we add `IsEnabled` to `TopLevel`?

**Comment by MarchingCube** [comment]
An interesting finding - it seems that several apps in Windows do support drag&drop for disabled windows. One example is Explorer: opening a modal dialog (Folder options for example) does not block dropping files.

File: src/Avalonia.Controls/TopLevel.cs [hunk 1]
```diff
@@ -285,6 +285,9 @@ IInputElement IInputRoot.PointerOverElement
         /// <inheritdoc/>
         IMouseDevice IInputRoot.MouseDevice => PlatformImpl?.MouseDevice;
 
+        /// <inheritdoc/>
+        bool? IInputRoot.IsPlatformEnabled => PlatformImpl is IWindowImpl windowImpl? windowImpl.IsEnabled : (bool?)null;
```
Comment (grokys) [inline src/Avalonia.Controls/TopLevel.cs:289]: Could be shortened to `(PlatformImpl as IWindowImpl)?.IsEnabled` I think?

File: src/Avalonia.Native/WindowImpl.cs [hunk 2]
```diff
@@ -220,5 +220,8 @@ public void SetEnabled(bool enable)
         {
             _native.SetEnabled(enable.AsComBool());
         }
+        
+        // TODO: Implement
+        public bool IsEnabled => throw new NotImplementedException();
```
Comment (grokys) [inline src/Avalonia.Native/WindowImpl.cs:225]: Does this need to be implemented in this PR? I think throwing here will currently break DnD on OSX, correct?

**Review by grokys — COMMENTED** [review]
LGTM, I find it a bit strange that we have the concept of "platform enabled" as well as the standard Avalonia `IsEnabled` but that was here before this PR.

Am I correct in thinking this is still a WIP due to the TODO in Avalonia.Native? We should make it as a draft/WIP if so.

File: src/Avalonia.Controls/TopLevel.cs (hunk 1 above)
Comment (mat1jaczyyy) [reply src/Avalonia.Controls/TopLevel.cs:289]: can do, yeah

File: src/Avalonia.Native/WindowImpl.cs (hunk 2 above)
Comment (mat1jaczyyy) [reply src/Avalonia.Native/WindowImpl.cs:225]: yes, i've no experience going low level with OSX so i'd rather someone else wrote this part

Comment (grokys) [reply src/Avalonia.Native/WindowImpl.cs:225]: Oh yeah sorry that's in the description. Need to learn to read.

**Comment by jmacato** [comment]
Closing this due to inactivity. Please ping me if you want to work on this again.

**Pull request closed without merging**